subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/internal/PmdReportsImpl.java

```diff
@@ -43,4 +46,19 @@ public SingleFileReport getHtml() {
     public SingleFileReport getXml() {
         return getByName("xml");
     }
+
+    @Override
+    public SingleFileReport getCsv() {
+        return getByName("csv");
+    }
+
+    @Override
+    public SingleFileReport getCodeClimate() {
+        return getByName("codeclimate");
+    }
+
+    @Override
+    public SingleFileReport getSarif() {
+        return getByName("sarif");
+    }
```

Comment:
The file names produced by these are not very nice (`main.codelimate`, `main.sarif`). Is there anything we can do about that? I tried using `codeclimate.json` and `sarif.json` instead, but that didn't work at all, it just gave me an error like this:

```
* What went wrong:
A problem occurred evaluating script.
> SingleFileReport with name 'sarif.json' not found.
```

Reply:
If you change the names of what you use to `add` it above, you can use `codeclimate.json` etc here instead.